ethereum/core/src/main/java/org/hyperledger/besu/ethereum/ConsensusContext.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum;
+
+@FunctionalInterface
+public interface ConsensusContext {
+  <C extends ConsensusContext> C get(final Class<C> klass);
```

Comment:
Is there a different name that can be used rather than `get`?  perhaps `getContext` to make it clear it's not something arbitrary or even `as` to make it clear it's almost a transformation?  or `castTo` to make it clear we are just doing a class cast?